### PR TITLE
Added onBlur to store target name change

### PR DESCRIPTION
**Pull request by diego-ge**
It was added an onBlur callback to store the target name.
Target name is being saved to DB on search and blur, but left side menu is not being updated properly.

### CODE REVIEW

File: explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala [hunk 1]
```diff
@@ -273,6 +273,7 @@ object SiderealTargetEditor {
                   // SearchForm doesn't edit the name directly. It will set it atomically, together
                   // with coords & magnitudes from the catalog search, so that all 3 fields are
                   // a single undo/redo operation.
+                  allView,
```
Comment (cquiroz) [inline explore/src/main/scala/explore/targeteditor/SiderealTargetEditor.scala:276]: maybe `allView` can replace the use of `targetView` as well, but we can take a look later

File: explore/src/main/scala/explore/targeteditor/SearchForm.scala [hunk 2]
```diff
@@ -119,6 +123,11 @@ object SearchForm {
           disabled = disabled,
           errorClazz = ExploreStyles.InputErrorTooltipBelow,
           errorPointing = LabelPointing.Above,
+          onBlur = (r: ValidatedNec[NonEmptyString, NonEmptyString]) =>
+            r match {
+              case Valid(name)   => props.targetView.zoom(Target.Sidereal.name).set(name)
+              case Invalid(name) => Callback.log(name)
+            },
```
Comment (toddburnside) [inline explore/src/main/scala/explore/targeteditor/SearchForm.scala:130]: Could this be done with a `withOnMod` on the view being used for the `value`?

Comment (toddburnside) [reply explore/src/main/scala/explore/targeteditor/SearchForm.scala:130]: I don't follow...

File: explore/src/main/scala/explore/targeteditor/SearchForm.scala [hunk 3]
```diff
@@ -119,6 +123,11 @@ object SearchForm {
           disabled = disabled,
           errorClazz = ExploreStyles.InputErrorTooltipBelow,
           errorPointing = LabelPointing.Above,
+          onBlur = (r: ValidatedNec[NonEmptyString, NonEmptyString]) =>
+            r match {
+              case Valid(name)   => props.targetView.zoom(Target.Sidereal.name).set(name)
+              case Invalid(name) => Callback.log(name)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/targeteditor/SearchForm.scala:129]: Is there a value to logging just the name when it is invalid?

File: explore/src/main/scala/explore/targeteditor/SearchForm.scala (hunk 2 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/SearchForm.scala:130]: In the props we don't have a `View[NonEmptyString]` of the name if we wanted to set it, only the concrete name

Comment (toddburnside) [reply explore/src/main/scala/explore/targeteditor/SearchForm.scala:130]: I was thinking of something like `value = View.fromState($).zoom(State.searchTerm).withOnMod(name => props.targetView.zoom(Target.Sidereal.name).set(name))`. Or something like that. I think it does the same thing as you have here. IIRC, the view is only changed on Valid so you don't need to worry about that. But maybe it isn't any better?

Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/SearchForm.scala:130]: I see, that's right  we could that

**Review by toddburnside — APPROVED** [review]
LGTM

**Comment by diego-ge** [comment]
The just the name field will be put in the undo stack and will be sent to the back end as it was done with the onBlur.
In case of a submit and a successful search the whole target will be put in the undo stack and will be sent to the backend.

If the name is modified and the user push enter or clicks on the submit button, both events (onBlur and submit) will be triggered, then the undo stack will have the whole target, as the last change and before that, will also have just the target name.

**Pull request merged**